gwlfe/ReadGwlfDataFile.py

```diff
@@ -3,9 +3,424 @@
+            z.NGLoadN[a] = (z.NumAnimals[a] * z.AvgAnimalWt[a] / 1000) * z.AnimalDailyN[a] * 365
+            z.NGLoadP[a] = (z.NumAnimals[a] * z.AvgAnimalWt[a] / 1000) * z.AnimalDailyP[a] * 365
+            z.NGLoadFC[a] = (z.NumAnimals[a] * z.AvgAnimalWt[a] / 1000) * z.FCOrgsPerDay[a] * 365
+            z.InitNgN += z.NGLoadN[a]
+            z.InitNgP += z.NGLoadP[a]
+            z.InitNgFC += z.NGLoadFC[a]
+        elif z.GrazingAnimal[a] is YesOrNo.YES:
+            z.GRLoadN[a] = (z.NumAnimals[a] * z.AvgAnimalWt[a] / 1000) * z.AnimalDailyN[a] * 365
+            z.GRLoadP[a] = (z.NumAnimals[a] * z.AvgAnimalWt[a] / 1000) * z.AnimalDailyP[a] * 365
+            z.GRLoadFC[a] = (z.NumAnimals[a] * z.AvgAnimalWt[a] / 1000) * z.FCOrgsPerDay[a] * 365
+            z.InitGrN += z.GRLoadN[a]
+            z.InitGrP += z.GRLoadP[a]
+            z.InitGrFC += z.GRLoadFC[a]
+        else:
+            raise ValueError('Unexpected value for GrazingAnimal')
+
+    # Obtain AEU for each farm animal
+    z.AEU1 = ((z.NumAnimals[3] / 2) * (z.AvgAnimalWt[3]) / 1000) + ((z.NumAnimals[4] / 2) * (z.AvgAnimalWt[4]) / 1000)
+    z.AEU2 = (z.NumAnimals[8] * z.AvgAnimalWt[8]) / 1000
+    z.AEU3 = (z.NumAnimals[6] * z.AvgAnimalWt[6]) / 1000
+    z.AEU4 = (z.NumAnimals[5] * z.AvgAnimalWt[5]) / 1000
+    z.AEU5 = (z.NumAnimals[7] * z.AvgAnimalWt[7]) / 1000
+    z.AEU6 = (z.NumAnimals[1] * z.AvgAnimalWt[1]) / 1000
+    z.AEU7 = (z.NumAnimals[2] * z.AvgAnimalWt[2]) / 1000
+
+    # Get the total AEU, Total livestock and poultry AEU
+    z.TotAEU = z.AEU1 + z.AEU2 + z.AEU3 + z.AEU4 + z.AEU5 + z.AEU6 + z.AEU7
+    z.TotLAEU = z.AEU3 + z.AEU4 + z.AEU5 + z.AEU6 + z.AEU7
+    z.TotPAEU = z.AEU1 + z.AEU2
+
+    # Get the Non-Grazing Animal Worksheet values
+    for i in range(12):
+        # For Non-Grazing
+        z.NGAccManAppN[i] += (z.InitNgN / 12) - (z.NGPctManApp[i] * z.InitNgN)
+
+        if z.NGAccManAppN[i] < 0:
+            z.NGAccManAppN[i] = 0
+
+        z.NGAccManAppP[i] += (z.InitNgP / 12) - (z.NGPctManApp[i] * z.InitNgP)
+
+        if z.NGAccManAppP[i] < 0:
+            z.NGAccManAppP[i] = 0
+
+        z.NGAccManAppFC[i] += (z.InitNgFC / 12) - (z.NGPctManApp[i] * z.InitNgFC)
+
+        if z.NGAccManAppFC[i] < 0:
+            z.NGAccManAppFC[i] = 0
+
+        z.NGAppManN[i] = z.NGPctManApp[i] * z.InitNgN
+        z.NGInitBarnN[i] = z.NGAccManAppN[i] - z.NGAppManN[i]
+
+        if z.NGInitBarnN[i] < 0:
+            z.NGInitBarnN[i] = 0
+
+        z.NGAppManP[i] = z.NGPctManApp[i] * z.InitNgP
+        z.NGInitBarnP[i] = z.NGAccManAppP[i] - z.NGAppManP[i]
+
+        if z.NGInitBarnP[i] < 0:
+            z.NGInitBarnP[i] = 0
+
+        z.NGAppManFC[i] = z.NGPctManApp[i] * z.InitNgFC
+        z.NGInitBarnFC[i] = z.NGAccManAppFC[i] - z.NGAppManFC[i]
+
+        if z.NGInitBarnFC[i] < 0:
+            z.NGInitBarnFC[i] = 0
+
+    # Read the Grazing Animal Worksheet values
+    for i in range(12):
+        z.GrazingN[i] = z.PctGrazing[i] * (z.InitGrN / 12)
```

Comment:
Yeah, that *is* confusing. In `GWLF-E.vbp` the file `ReadAllDataFiles.bas` is aliased to `ReadGwlfDataFile`...

```
Module=ReadGwlfDataFile; ReadAllDataFiles.bas
```

I could rename this to `ReadAllDataFiles.bas` to avoid confusion. On the other hand, this module no longer "reads" data files, it just prepares a few calculations before running the main model. Neither `ReadAllDataFiles` nor `ReadGwlfDataFile` really describes what this code does very well.